local_build/local_build.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Locally build and compress the core Blockly files into a single JavaScript
+# file.
+#
+# Copyright 2018 Google Inc.
+# https://developers.google.com/blockly/
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Usage: local_build.sh.
+#
+# This script generates only local_blockly_compressed_vertical.js.  You may
+# modify it as needed to build other files.
+#
+# The compressed file is a concatenation of all of Scratch-Blocks's core files,
+# run through a local copy of Google's Closure Compiler with simple
+# optimizations turned on.
+
+# Future work:
+# - Trim down Google's Apache licenses, to match the output of build.py.
+# - Generate other compressed files generated by build.py normally.
+# - Add a good error message if multiple versions of the closure compiler were
+#   found.
+# - Add a flag for generating horizontal (still default to vertical).
+
+# Find the Closure Compiler.
+if [ -f "$(npm root)/google-closure-compiler/compiler.jar" ]; then
+  COMPILER="$(npm root)/google-closure-compiler/compiler.jar"
+elif [ -f closure-compiler*.jar ]; then
+  COMPILER="closure-compiler*.jar"
+  # TODO: Check whether multiple files were found.
+else
+  echo "ERROR: Closure Compiler not found."
+  echo "Download from this URL, and place jar file in current directory."
+  echo "https://dl.google.com/closure-compiler/compiler-latest.zip"
+  exit 1
+fi
+
+echo Using $COMPILER as the compiler.
+rm local_blockly_compressed_vertical.js 2> /dev/null
+echo Compiling Scratch-Blocks..
+java -jar $COMPILER \
+  --js='../core/**.js' \
```

Comment:
Is there a reason we should have this inside an enclosing `local_build` directory? My instinct would be to have this at the top level along with the `build.py` file.

Reply:
I agree that putting it at the top level would match what we're already doing, but I'm not stoked on how many files are at the top level.  Moving the existing files seems like a pain, so I'm just trying to avoid adding new ones.  

This adds three (closure-compiler, the built file, and this file).  

If you want it at the top level, I'll move it.  Just let me know.

Reply:
I think I'd rather move it to the top level (even though I agree it's pretty crowded). Needing to `cd` into `local_build` feels like it may cause confusion.